src/main/java/chess/controller/ExceptionController.java

```diff
@@ -0,0 +1,28 @@
+package chess.controller;
+
+import chess.exception.IllegalGameProgressException;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+@ControllerAdvice
+public class ExceptionController {
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    public String handleMove(RedirectAttributes redirectAttributes, IllegalArgumentException e, HttpServletRequest request) {
+        redirectAttributes.addAttribute("error", e.getMessage());
+
+        String referer = request.getHeader("Referer");
+        String[] split = referer.split("\\?");
+
+        return "redirect:" + split[0];
```

Comment:
```suggestion
        String referer = request.getHeader("Referer");
        String previousPath = URI.create(referer)
            .getPath();

        return "redirect:" + previousPath;
```
기존 자바패키지에 제공하는 기능을 활용해보는건 어떨까요?

Reply:
URI.create() 을 통해서 인자로 전달한 문자열로 URI 객체를 생성하고, getPath() 를 이용해서 해당 URI의 디코딩된 경로만을 반환받음으로써 쿼리스트링의 중첩을 막을 수 있겠군요! 감사합니다!!